inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/management/DynamicWorkloadManagementPage.java

```diff
@@ -342,33 +356,253 @@ protected void onUpdate(AjaxRequestTarget ajaxRequestTarget) {
 
-        //Submit button
-        Button search = new LambdaAjaxButton("search", this::actionSubmit);
+        // Submit button
+        Button search = new LambdaAjaxButton<>("search", this::actionSubmit);
 
         searchForm.add(search);
 
-        //Condition for filter inputs to be enabled
-        dateTo.add(enabledWhen(() ->
-            !dateChoices.getValue().equals(dateChoice.get(0))
-            && unused.getValue().equals("false")));
-        dateFrom.add(enabledWhen(() ->
-            !dateChoices.getValue().equals(dateChoice.get(1))
-            && unused.getValue().equals("false")));
-        dateChoices.add(enabledWhen(() ->
-            unused.getValue().equals("false")));
-        userFilterTextField.add(enabledWhen(() ->
-            unused.getValue().equals("false")));
-        documentFilterTextField.add(enabledWhen(() ->
-            unused.getValue().equals("false")));
+        // Condition for filter inputs to be enabled
+        dateTo.add(enabledWhen(
+            () -> !dateChoices.getDefaultModelObjectAsString().equals(dateChoice.get(0))
+                 && unused.getValue().equals("false")));
+        dateFrom.add(enabledWhen(
+            () -> !dateChoices.getDefaultModelObjectAsString().equals(dateChoice.get(1))
+                 && unused.getValue().equals("false")));
+        dateChoices.add(enabledWhen(() -> unused.getValue().equals("false")));
+        userFilterTextField.add(enabledWhen(() -> unused.getValue().equals("false")));
+        documentFilterTextField.add(enabledWhen(() -> unused.getValue().equals("false")));
 
         return searchForm;
     }
 
+    public Form<Void> createSettingsForm()
+    {
+        Form<Void> settingsForm = new Form<>("settingsForm");
+        settingsForm.setOutputMarkupId(true);
+
+        // Init defaultDocumentsNumberTextField
+        defaultNumberDocumentsTextField = new NumberTextField<>("defaultDocumentsNumberTextField",
+                new Model<>(), Integer.class);
+
+        WorkloadTraits traits = dynamicWorkloadExtension.readTraits(
+            workloadManagementService.getOrCreateWorkloadManagerConfiguration(
+                currentProject.getObject()));
+
+        defaultNumberDocumentsTextField.setDefaultModel(new CompoundPropertyModel<>(
+            (traits.getDefaultNumberOfAnnotations())));
+        defaultNumberDocumentsTextField.setMinimum(1);
+        defaultNumberDocumentsTextField.setRequired(true);
+        defaultNumberDocumentsTextField.setConvertEmptyInputStringToNull(false);
+
+        // Finally, add the confirm button at the end
+        Button save = new LambdaAjaxButton<>("save", this::actionConfirm);
+
+        settingsForm.add(defaultNumberDocumentsTextField);
+        settingsForm.add(save);
+
+        workflowChoices = new BootstrapSelect<>("workloadStrategy");
+        workflowChoices.setDefaultModel(Model.of(traits.getWorkloadType()));
+        List<String> choices = new ArrayList<>();
+        choices.add(getString("def"));
```

Comment:
Not good. Looks like you want to have an extensible list here, so you have to introduce an extension point where these options can be plugged in...

@jcklie Can you please have an eye on this?